src/main/java/uk/gov/pay/api/filter/RateLimiterKey.java

```diff
@@ -0,0 +1,46 @@
+package uk.gov.pay.api.filter;
+
+import uk.gov.pay.api.utils.PathHelper;
+
+import javax.ws.rs.container.ContainerRequestContext;
+
+public class RateLimiterKey {
+    
+    private String method;
+    private String key;
+    private String keyType;
+
+    public RateLimiterKey(String key, String keyType, String method) {
+        this.key = key;
+        this.keyType = keyType;
+        this.method = method;
+    }
+
+    public static RateLimiterKey from(ContainerRequestContext requestContext) {
+        final String method = requestContext.getMethod();
+
+        StringBuilder builder = new StringBuilder(method);
+
+        final String pathType = PathHelper.getPathType(requestContext.getUriInfo().getPath(), method);
+        if (!pathType.isBlank()) {
+            builder.append("-" + pathType);
+        }
+
+        final String keyType = builder.toString();
+        builder.append("-" + requestContext.getHeaderString("Authorization"));
```

Comment:
I realise you didn't introduce this idea, but I'm not sure about using the raw API key as part of the redis key? Can we not make a cheap hash of it ore something?

Reply:
a further thought - it looks like you have access to the account id when creating the rate limit key, so why not use that rather than api key?

Reply:
🤷‍♀ 
I didn't want to change the existing logic. Happy to change to account id if that's preferable.

Reply:
@kbottla any thoughts on this - was there a reason to use api key rather than account id?

Reply:
I've changed it to account id (in the second commit)

Reply:
@mrlumbu We have used API key before as account id was not available and missed on hashing API Key. Now that it is available we can use it for key.